builtin/providers/fastly/resource_fastly_service_v1.go

```diff
@@ -0,0 +1,603 @@
+	// If `force_destroy` is given, we deactivate the active version and then send
+	// the DELETE call
+	if d.Get("force_destroy").(bool) {
+		s, err := conn.GetServiceDetails(&gofastly.GetServiceInput{
+			ID: d.Id(),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if s.ActiveVersion.Number != "" {
+			_, err := conn.DeactivateVersion(&gofastly.DeactivateVersionInput{
+				Service: d.Id(),
+				Version: s.ActiveVersion.Number,
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err := conn.DeleteService(&gofastly.DeleteServiceInput{
+		ID: d.Id(),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	_, err = findService(d.Id(), meta)
+	if err != nil {
+		switch err {
+		// we expect no records to be found here
+		case fastlyNoServiceFoundErr:
+			d.SetId("")
+			return nil
+		default:
+			return err
+		}
+	}
+
+	// findService above returned something and nil error, but shouldn't have
+	return fmt.Errorf("[WARN] Tried deleting Service (%s), but was still found", d.Id())
+
+}
+
+func flattenDomains(list []*gofastly.Domain) []map[string]interface{} {
+	dl := make([]map[string]interface{}, 0, len(list))
+
+	for _, d := range list {
+		dl = append(dl, map[string]interface{}{
+			"name":    d.Name,
+			"comment": d.Comment,
+		})
+	}
+
+	return dl
+}
+
+func flattenBackends(backendList []*gofastly.Backend) []map[string]interface{} {
+	var bl []map[string]interface{}
+	for _, b := range backendList {
+		// Convert Backend to a map for saving to state.
+		nb := map[string]interface{}{
+			"name":                  b.Name,
+			"address":               b.Address,
+			"auto_loadbalance":      b.AutoLoadbalance,
+			"between_bytes_timeout": int(b.BetweenBytesTimeout),
+			"connect_timeout":       int(b.ConnectTimeout),
+			"error_threshold":       int(b.ErrorThreshold),
+			"first_byte_timeout":    int(b.FirstByteTimeout),
+			"max_conn":              int(b.MaxConn),
+			"port":                  int(b.Port),
+			"ssl_check_cert":        b.SSLCheckCert,
+			"weight":                int(b.Weight),
+		}
+
+		bl = append(bl, nb)
+	}
+	return bl
+}
+
+// findService finds a Fastly Service via the ListServices endpoint, returning
+// the Service if found.
+//
+// Fastly API does not include any "deleted_at" type parameter to indicate
+// that a Service has been deleted. GET requests to a deleted Service will
+// return 200 OK and have the full output of the Service for an unknown time
+// (days, in my testing). In order to determine if a Service is deleted, we
+// need to hit /service and loop the returned Services, searching for the one
+// in question. This endpoint only returns active or "alive" services. If the
+// Service is not included, then it's "gone"
+//
+// Returns a fastlyNoServiceFoundErr error if the Service is not found in the
+// ListServices response.
+func findService(id string, meta interface{}) (*gofastly.Service, error) {
+	conn := meta.(*FastlyClient).conn
+
+	l, err := conn.ListServices(&gofastly.ListServicesInput{})
```

Comment:
Seems like there must be a way for us to find a specific service rather than having to pull down all services in an account and page through them in memory?

This seems like it could get hairy for larger Fastly accounts.